dashbuilder/dashbuilder-shared/dashbuilder-js/packages/component-api/src/controller/BrowserComponentBus.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ComponentMessage, MessageProperty } from "../message";
+import { ComponentBus } from "./ComponentBus";
+
+export class BrowserComponentBus implements ComponentBus {
+  private componentId: string;
+
+  private listener: (message: ComponentMessage) => void;
+
+  private readonly messageListener = (e: MessageEvent) => {
+    this.listener(e.data as ComponentMessage);
+  };
+
+  public start() {
+    window.addEventListener("message", this.messageListener, false);
+  }
+
+  public send(message: ComponentMessage): void {
+    console.debug("[BrowserComponentBus] Sending Message");
+    console.debug(message);
+    message.properties.set(MessageProperty.COMPONENT_ID, this.componentId);
```

Comment:
The issue is that we need to receive the componentid from dashbuilder so we can identify the component itself so dashbuilder can send messages to it. This way we can reuse the same message bus for all components.

What I changed was to keep the componentId on controller - it may be undefined at the beginning, but once INIT is received the controller will get it.

Feel free to suggest any other kind of change related to this!